### PR TITLE
Add spock integration testing to the project and gradle-plugin module.

### PR DESCRIPTION
**Pull request by Astro03**
This script Create a script to allow spock integration testing.

As per https://guides.gradle.org/testing-gradle-plugins/#integration-tests:
Integration tests or "functional tests verify the correctness of the [Gradle]
plugin end-to-end. In practice that means applying, configuring and
executing the functionality of the plugin implementation".

* Create a script that adds spock integration testing for a module
* Add integration testing to the gradle-plugin module
* Create an integ test that validates the pegasus plugin can be applied

### CODE REVIEW

File: gradle-plugins/build.gradle [hunk 1]
```diff
@@ -1,10 +1,9 @@
-apply plugin: 'java'
+// Setup integTests
+apply from: "${buildScriptDirPath}/integTest.gradle"
 
 dependencies {
-  compile gradleApi()
```
Comment (DPUkyle) [inline gradle-plugins/build.gradle:4]: NB: `compile gradleApi()` is applied automatically by the `java-gradle-plugin`

Comment (Astro03) [reply gradle-plugins/build.gradle:4]: Exactly. That is why i put that note in "integTest.gradle"

File: gradle-plugins/src/integTest/groovy/com/linkedin/pegasus/gradle/PegasusPluginIntegrationTest.groovy [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    when:
+    def result = GradleRunner.create()
+        .withProjectDir(tempDir.root)
+        .withPluginClasspath()
+        .withArguments("mainDataTemplateJar")
+        .forwardOutput()
+        .build()
+
+    then:
+    result.output.contains("BUILD SUCCESSFUL")
```
[the remark this answers is not in the document]
Comment (Astro03) [reply gradle-plugins/src/integTest/groovy/com/linkedin/pegasus/gradle/PegasusPluginIntegrationTest.groovy]: done.. thanks

File: build.gradle [hunk 3]
```diff
@@ -286,10 +287,11 @@ subprojects {
     }
   }
 
-  // Exclude tests which are known to be flaky in the Travis CI environment
+  // Exclude tests which are known to be flaky in the Travis CI environment. Since all test tasks are TestNG tasks
+  // EXCEPT the integTest task, exclude the options from being evaluated by travis for the integTest task.
   if (System.getenv('TRAVIS') == 'true' && System.getenv('USER') == 'travis') {
     afterEvaluate {
-      project.tasks.withType(Test).forEach {
+      project.tasks.withType(Test).matching { it.name != "integTest" }.forEach {
```
Comment (DPUkyle) [inline build.gradle:294]: I suggest:
```
project.tasks.withType(Test) {
  if (it.options instanceof TestNGOptions) {
    it.options.excludeGroups 'ci-flaky'
    it.systemProperties['test.httpRequestTimeout'] = '20000'
  }
}
```

Comment (Astro03) [reply build.gradle]: good idea, it's better than what I was doing.

**Comment by DPUkyle** [comment]
> Looks like this is just setting up the integration testing without testing any specific pegasus behavior just yet, so I'll trust yours and Kyle's opinions on this. LGTM, thanks for the contribution

That's right. #337 will add more meaningful tests.

**Pull request merged**